gprofiler/merge.py

```diff
@@ -332,6 +333,46 @@ def _enrich_and_finalize_stack(
     return f"{enrich_data.application_prefix}{enrich_data.container_prefix}{stack} {count}"
 
 
+def _merge_metadata(primary, secondary):
+    for key, value in primary.items():
+        # in case value is a dict itself
+        if isinstance(value, dict):
+            node = secondary.setdefault(key, {})
+            _merge_metadata(value, node)
+        else:
+            secondary[key] = value
+    return secondary
+
+
+def concatenate_from_external_file(
+    collapsed_file_path: str,
+    obtained_metadata: Metadata,
+) -> Tuple[str, str, str, int]:
+    """
+    Concatenate all stacks from all stack mappings in process_profiles.
+    Add "profile metadata" and metrics as the first line of the resulting collapsed file.
+    """
+    total_samples = 0
+    lines = []
+    start_time = None
+    end_time = None
+
+    # TODO: container names and application metadata
+    with open(collapsed_file_path) as file:
+        for index, line in enumerate(file):
+            if index == 0:
+                read_metadata = json.loads(line[2:])
```

Comment:
Check if line begins with comment (i.e `#`, and then the json itself can be immediately after it `#{...}` or with a space `# {}`. I suggest you just `json.loads(line[1:])` and it'll work for both cases